### PR TITLE
Fix/wipe vital app revision

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

As our request reschedule flow got upgraded, our wipemycal/vital app that was sharing same code base should be updated as well. It no longer create a new event in calendar with the legend Cancelled.

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Should not recreate calendar events on users account, just delete what it was rescheduled.

### CODE REVIEW

File: apps/web/pages/api/book/request-reschedule.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ const handler = async (
     if (session?.user?.id) {
       userOwner = await findUserDataByUserId(session?.user.id);
     } else {
-      return res.status(501);
+      return res.status(501).end();
```
Comment (alannnc) [inline apps/web/pages/api/book/request-reschedule.ts:64]: Prevent stalled request

File: apps/web/playwright/fixtures/bookings.ts [hunk 2]
```diff
@@ -21,17 +21,19 @@ export const createBookingsFixture = (page: Page) => {
       userId: number,
       username: string | null,
       eventTypeId = -1,
-      { rescheduled = false, paid = false, status = "ACCEPTED" }: Partial<Prisma.BookingCreateInput> = {}
+      { rescheduled = false, paid = false, status = "ACCEPTED" }: Partial<Prisma.BookingCreateInput> = {},
+      startDateParam?: Date,
+      endDateParam?: Date
```
Comment (alannnc) [inline apps/web/playwright/fixtures/bookings.ts:26]: Added to params to createBookings with custom startDate, endDate

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+test.describe("Validate Wipe my Calendar button", () => {
+  test.use({ storageState: "playwright/artifacts/proStorageState.json" });
+  test("Browse upcoming bookings and validate button shows", async ({ page, users, bookings }) => {
+    const pro = await users.create();
+    const [eventType] = pro.eventTypes;
+    await prisma?.credential.create({
+      data: {
+        key: {},
+        type: "wipemycal_other",
+        userId: pro.id,
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:17]: `appId` is missing here

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      pro.id,
+      pro.username,
+      eventType.id,
+      {},
+      dayjs().endOf("day").subtract(29, "minutes").toDate(),
+      dayjs().endOf("day").toDate()
+    );
+    await pro.login();
+    await page.goto("/bookings/upcoming");
+    await expect(page.locator("data-testid=wipe-today-button")).toBeVisible();
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:30]: Let's cleanup here: `await users.deleteAll()`

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    // eslint-disable-next-line playwright/no-wait-for-timeout
+    await page.waitForTimeout(200);
+    const totalUserBookingsCancelled = await prisma?.booking.findMany({
+      where: {
+        userId: pro.id,
+        status: "CANCELLED",
+      },
+    });
+
+    await expect(totalUserBookingsCancelled?.length).toBe(1);
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:76]: Cleanup here as well: `await users.deleteAll()`

File: packages/app-store/vital/lib/reschedule.ts [hunk 6]
```diff
@@ -139,19 +139,6 @@ const Reschedule = async (bookingUid: string, cancellationReason: string) => {
...
-        credentials: userOwner.credentials,
-        destinationCalendar: userOwner.destinationCalendar,
-      });
-      builder.calendarEvent.title = `Cancelled: ${builder.calendarEvent.title}`;
-      await eventManager.updateAndSetCancelledPlaceholder(builder.calendarEvent, bookingToReschedule);
-    } catch (error) {
-      if (error instanceof Error) {
-        logger.error(error.message);
-      }
-    }
```
Comment (zomars) [inline packages/app-store/vital/lib/reschedule.ts:154]: Is this intended?

**Review by zomars — COMMENTED** [review]
Changes look good I'm just having a hard time understanding what is going on.

File: packages/app-store/vital/lib/reschedule.ts (hunk 6 above)
Comment (alannnc) [reply packages/app-store/vital/lib/reschedule.ts:154]: Yup, this code was shared with request-reschedule from our main api, and got removed there and should be removed here as well.

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts (hunk 3 above)
Comment (alannnc) [reply apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:17]: Thanks!

**Comment by alannnc** [comment]
> Changes look good I'm just having a hard time understanding what is going on.

Just updated with suggested changes, would edit PR description to make it more clear what is happening

**Review by pumfleet — APPROVED** [review]
Looks good to me, haven't tested yet

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    users,
+    bookings,
+  }) => {
+    const pro = await users.create();
+    const [eventType] = pro.eventTypes;
+    await prisma?.credential.create({
+      data: {
+        key: {},
+        type: "wipemycal_other",
+        userId: pro.id,
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:20]: appId is still missing here

```suggestion
        userId: pro.id,
        appId: "wipemycal",
```

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+test.describe("Wipe my Cal App Test", () => {
+  test("Browse upcoming bookings and validate button shows and triggering wipe my cal button", async ({
+    page,
+    users,
+    bookings,
+  }) => {
+    const pro = await users.create();
+    const [eventType] = pro.eventTypes;
+    await prisma?.credential.create({
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:16]:
```
import prisma from "@calcom/prisma";
```

```suggestion
    await prisma.credential.create({
```

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    });
+    await bookings.create(
+      pro.id,
+      pro.username,
+      eventType.id,
+      {},
+      dayjs().endOf("day").subtract(29, "minutes").toDate(),
+      dayjs().endOf("day").toDate()
+    );
+    await prisma?.credential.create({
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:31]: same here

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+      },
+    });
+
+    await bookings.create(pro.id, pro.username, eventType.id, {});
+    await bookings.create(pro.id, pro.username, eventType.id, {});
+    await pro.login();
+    await page.goto("/bookings/upcoming");
+    await expect(page.locator("data-testid=wipe-today-button")).toBeVisible();
+
+    const totalUserBookings = await prisma?.booking.findMany({
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:46]:
```suggestion
    const totalUserBookings = await prisma.booking.findMany({
```

File: apps/web/playwright/lib/appstore/wipe-my-cal.test.ts [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+      where: {
+        userId: pro.id,
+      },
+    });
+    expect(totalUserBookings?.length).toBe(3);
+    await page.locator("data-testid=wipe-today-button").click();
+    await page.locator("data-testid=send_request").click();
+    // eslint-disable-next-line playwright/no-wait-for-timeout
+    await page.waitForTimeout(250);
+    const totalUserBookingsCancelled = await prisma?.booking.findMany({
```
Comment (zomars) [inline apps/web/playwright/lib/appstore/wipe-my-cal.test.ts:56]:
```suggestion
    const totalUserBookingsCancelled = await prisma.booking.findMany({
```

**Review by alannnc — COMMENTED** [review]
@zomars All comments should done, thank you for your suggestions.

**Review by zomars — APPROVED** [review]
Let's go!

**Pull request merged**